cmd/kube-apiserver/app/options/options_test.go

```diff
@@ -17,26 +17,164 @@ limitations under the License.
-	}
 
 	args := []string{
+		"--advertise-address=192.168.10.10",
+		"--allow-privileged=false",
+		"--anonymous-auth=false",
+		"--apiserver-count=5",
+		"--audit-log-maxage=11",
+		"--audit-log-maxbackup=12",
+		"--audit-log-maxsize=13",
+		"--audit-log-path=/var/log",
+		"--audit-policy-file=/policy",
+		"--audit-webhook-config-file=/webhook-config",
+		"--audit-webhook-mode=blocking",
+		"--authentication-token-webhook-cache-ttl=3m",
+		"--authentication-token-webhook-config-file=/token-webhook-config",
+		"--authorization-mode=AlwaysDeny",
+		"--authorization-policy-file=/policy",
+		"--authorization-webhook-cache-authorized-ttl=3m",
+		"--authorization-webhook-cache-unauthorized-ttl=1m",
+		"--authorization-webhook-config-file=/webhook-config",
+		"--bind-address=192.168.10.20",
+		"--client-ca-file=/client-ca",
+		"--cloud-config=/cloud-config",
+		"--cloud-provider=daocloud",
+		"--cors-allowed-origins=10.10.10.100,10.10.10.200",
+		"--enable-aggregator-routing=true",
+		"--enable-logs-handler=false",
 		"--enable-swagger-ui=true",
 	}
 	f.Parse(args)
-	if !s.Features.EnableSwaggerUI {
-		t.Errorf("Expected s.EnableSwaggerUI to be true")
+
+	// Values parsed by args
+	parsedOps := Options{}
+	parsedOps.AdvertiseAddress = s.GenericServerRunOptions.AdvertiseAddress
+	parsedOps.AllowPrivileged = s.AllowPrivileged
+	parsedOps.AnonymousAuth = s.Authentication.Anonymous.Allow
+	parsedOps.ApiserverCount = s.MasterCount
+	parsedOps.AuditLogMaxage = s.Audit.LogOptions.MaxAge
+	parsedOps.AuditLogMaxbackup = s.Audit.LogOptions.MaxBackups
+	parsedOps.AuditLogMaxsize = s.Audit.LogOptions.MaxSize
+	parsedOps.AuditLogPath = s.Audit.LogOptions.Path
+	parsedOps.AuditPolicyFile = s.Audit.PolicyFile
+	parsedOps.AuditWebhookConfigFile = s.Audit.WebhookOptions.ConfigFile
+	parsedOps.AuditWebhookMode = s.Audit.WebhookOptions.Mode
+	parsedOps.AuthenticationTokenWebhookCacheTTL = s.Authentication.WebHook.CacheTTL
+	parsedOps.AuthorizationWebhookConfigFile = s.Authentication.WebHook.ConfigFile
+	parsedOps.AuthorizationMode = s.Authorization.Mode
+	parsedOps.AuthorizationPolicyFile = s.Authorization.PolicyFile
+	parsedOps.AuthorizationWebhookCacheAuthorizedTTL = s.Authorization.WebhookCacheAuthorizedTTL
+	parsedOps.AuthorizationWebhookCacheUnauthorizedTTL = s.Authorization.WebhookCacheUnauthorizedTTL
+	parsedOps.AuthorizationWebhookConfigFile = s.Authorization.WebhookConfigFile
+	parsedOps.BindAddress = s.SecureServing.BindAddress
+	parsedOps.ClientCAFile = s.Authentication.ClientCert.ClientCA
+	parsedOps.CloudConfig = s.CloudProvider.CloudConfigFile
+	parsedOps.CloudProvider = s.CloudProvider.CloudProvider
+	parsedOps.CorsAllowedOrigins = s.GenericServerRunOptions.CorsAllowedOriginList
+	parsedOps.EnableAggregatorRouting = s.EnableAggregatorRouting
+	parsedOps.EnableLogsHandler = s.EnableLogsHandler
+	parsedOps.EnableSwaggerUI = s.Features.EnableSwaggerUI
+
+	// Values which parsed values should be equal to
+	testOps := Options{}
```

Comment:
Why do we need this special type? Why can't we just use the existing `Options` type, create the instance of that with the expected values and compare `s` with that?